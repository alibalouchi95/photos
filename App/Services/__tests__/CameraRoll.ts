import {
    getPhotos,
    chooseProfilePhoto,
    launchCamera,
    launchImageLibrary
} from '../CameraRoll'

describe('camera roll', () => {
    describe('getPhotos', () => {
      it('get an arry of uris', async () => {
          expect(getPhotos(1)).resolves.toMatchSnapshot()
      })
    })
    describe('launchCamera', () => {
        it('should successfully launch', async () => {
            const result = await launchCamera()
            expect(result).toMatchSnapshot()
        })
    })
    describe('launchImageLibrary', () => {
        it('should successfully launch', async () => {
            expect(launchImageLibrary()).resolves.toMatchSnapshot()
        })
    })
    describe('chooseProfilePhoto', () => {
        it('should get a valid photo response, user cancel, and error', async () => {
            // Success
            expect(chooseProfilePhoto()).resolves.toMatchSnapshot()
            // User cancel
            expect(chooseProfilePhoto()).rejects.toEqual(new Error('user cancelled'))
            // Camera error
            expect(chooseProfilePhoto()).rejects.toEqual(new Error('mock error'))
        })
    })
})
