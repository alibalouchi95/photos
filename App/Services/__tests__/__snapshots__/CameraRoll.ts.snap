// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`camera roll chooseProfilePhoto should get a valid phto response, user cancel, and error 1`] = `
Object {
  "data": undefined,
  "image": Object {
    "canDelete": true,
    "customButton": undefined,
    "didCancel": false,
    "error": undefined,
    "height": 200,
    "isVertical": false,
    "origURL": "uri",
    "path": "uri",
    "uri": "uri",
    "width": 200,
  },
}
`;

exports[`camera roll getPhotos get an arry of uris 1`] = `
Array [
  "/disk/uri",
]
`;

exports[`camera roll launchCamera should successfully launch 1`] = `
Object {
  "canDelete": true,
  "customButton": undefined,
  "didCancel": false,
  "error": undefined,
  "height": 200,
  "isVertical": false,
  "origURL": "uri",
  "path": "uri",
  "uri": "uri",
  "width": 200,
}
`;

exports[`camera roll launchImageLibrary should successfully launch 1`] = `
Object {
  "canDelete": true,
  "customButton": undefined,
  "didCancel": false,
  "error": undefined,
  "height": 200,
  "isVertical": false,
  "origURL": "uri",
  "path": "uri",
  "uri": "uri",
  "width": 200,
}
`;
